maplibre_gl_example/lib/map_snapshot.dart

```diff
@@ -0,0 +1,307 @@
+    canvas.drawRect(Rect.fromLTWH(flagX, flagY, stripeWidth, flagHeight), bluePaint);
+    
+    // White stripe
+    final whitePaint = Paint()..color = Colors.white;
+    canvas.drawRect(Rect.fromLTWH(flagX + stripeWidth, flagY, stripeWidth, flagHeight), whitePaint);
+    
+    // Red stripe
+    final redPaint = Paint()..color = const Color(0xFFEF4135);
+    canvas.drawRect(Rect.fromLTWH(flagX + stripeWidth * 2, flagY, stripeWidth, flagHeight), redPaint);
+    
+    // Draw border
+    final borderPaint = Paint()
+      ..color = Colors.black
+      ..style = PaintingStyle.stroke
+      ..strokeWidth = 1 * pixelRatio;
+    canvas.drawRect(Rect.fromLTWH(flagX, flagY, flagWidth, flagHeight), borderPaint);
+    
+    final picture = recorder.endRecording();
+    final image = await picture.toImage(size.toInt(), size.toInt());
+    final byteData = await image.toByteData(format: ui.ImageByteFormat.png);
+    
+    return byteData!.buffer.asUint8List();
+  }
+
+  Future<void> _takeSnapshot() async {
+    setState(() {
+      _isLoading = true;
+      _error = null;
+      _snapshotImage = null;
+      _currentCity = null;
+    });
+
+    try {
+      // Define random camera positions for different cities
+      final cities = [
+        {'name': 'Beijing', 'lat': 39.9042, 'lng': 116.4074},
+        {'name': 'San Francisco', 'lat': 37.7749, 'lng': -122.4194},
+        {'name': 'London', 'lat': 51.5074, 'lng': -0.1278},
+        {'name': 'Sydney', 'lat': -33.8688, 'lng': 151.2093},
+      ];
+      
+      // Pick a random city
+      final randomCity = cities[DateTime.now().millisecond % cities.length];
+      
+      final cameraPosition = CameraPosition(
+        target: LatLng(randomCity['lat'] as double, randomCity['lng'] as double),
+        zoom: _currentZoom,
+      );
+
+      // Take the snapshot with DPR consideration
+      final pixelRatio = MediaQuery.of(context).devicePixelRatio;
+      
+      // Create markers if enabled
+      List<MapMarker>? markers;
+      if (_showMarkers) {
+        final cityName = randomCity['name'] as String;
+
+        markers = [
+          MapMarker(
+            position: LatLng(randomCity['lat'] as double, randomCity['lng'] as double),
+            iconData: await _createMarkerIcon('📍', pixelRatio),
+            iconSize: 36.0,
+            label: cityName,
+          ),
+          MapMarker(
+            position: LatLng(48.8566, 2.3522),
+            iconData: await _createFrenchFlagIcon(pixelRatio),
+            iconSize: 24.0,
+            label: 'Paris',
+          ),
+        ];
+      }
+      final imageData = await startMapSnapshot(
+        width: (400 * pixelRatio).toInt(),
+        height: (300 * pixelRatio).toInt(),
+        styleUrl: 'https://demotiles.maplibre.org/style.json',
+        cameraPosition: cameraPosition,
+        markers: markers,
+      );
+
+      setState(() {
+        _snapshotImage = imageData;
+        _currentCity = randomCity['name'] as String;
```

Comment:
Adjust the casting type of the variable
```suggestion
        _currentCity = randomCity['name'] as String?;
```